FileCopy/utils.cpp

```diff
@@ -1,13 +1,52 @@
 #include "utils.h"
 
+#include <sys/stat.h>
+
+#include <cassert>
+#include <cstdlib>
 #include <fstream>
 
 #include "c150debug.h"
+#include "c150nastyfile.h"
 
 using namespace C150NETWORK;
 using namespace std;
 
-void setUpDebugLogging(const char *logname, int argc, char *argv[]) {
+bool check_directory(const char *dirname) {
+    assert(dirname);
+    struct stat statbuf;
+    if (lstat(dirname, &statbuf) != 0) {
+        fprintf(stderr, "Error stating supplied source directory %s\n",
+                dirname);
+        exit(8);
+    }
+
+    if (!S_ISDIR(statbuf.st_mode)) {
+        fprintf(stderr, "File %s exists but is not a directory\n", dirname);
+        exit(8);
+    }
+    return false;
+}
+
+bool is_file(const char *filename) {
+    assert(filename);
+    struct stat statbuf;
+    if (lstat(filename, &statbuf) != 0) {
+        fprintf(stderr, "is file: Error stating supplied source file %s\n",
+                filename);
+        return false;
+    }
+
+    if (!S_ISREG(statbuf.st_mode)) {
+        fprintf(stderr, "is file: %s exists but is not a regular file\n",
+                filename);
+        return false;
+    }
+
+    return true;
+}
```

Comment:
Similar thing as check_directory, we can combine the conditionals. We should not print to stderr here since it looks like this function can return false during normal operation? I think we could rewrite this as
```
bool is_file(const char *filename) {
     assert(filename);
     struct stat statbuf;
     return (lstat(filename, &statbuf) == 0) && S_ISREG(statbuf.st_mode);
 }
 ```